graylog2-web-interface/src/views/components/searchbar/QueryInput.jsx

```diff
@@ -104,4 +104,4 @@ QueryInput.defaultProps = {
   placeholder: '',
 };
 
-export default withPluginEntities(QueryInput, { completers: 'views.completers' });
+export default withTheme(withPluginEntities(QueryInput, { completers: 'views.completers' }));
```

Comment:
Initially I wrapped `QueryInput` with `withTheme`,  but it results in a flow error I can't fix easily:
```
Cannot call withPluginEntities because property completers is missing in object type [1] in the first argument.
[prop-missing]

     src/views/components/searchbar/QueryInput.jsx
     105│   placeholder: '',
     106│ };
     107│
     108│ export default withPluginEntities(withTheme(QueryInput), { completers: 'views.completers' });
     109│

     flow-typed/npm/styled-components_v4.x.x.js
 [1] 175│   declare export function withTheme<Theme, Config: {...}, Instance>(Component: React$AbstractComponent<Config, Instance>): React$AbstractComponent<$Diff<Config, ThemeProps<Theme | void>>, Instance>
```

I prefer the above structure and we do it similar in comparable cases where we use `withPluginEntities`  (e.g. in combination with `withRouter`), but I see no other disadvantage regarding the current structure.

Reply:
It is fine this way, imo. It should work both ways around though, so I took a quick look into the issue. There are two things: 

a) we are still using `styled-components` for v4. They do not work with functional components (like `QueryInput` is), so the type inference for the component does not work. I was able to force it by passing explicit types for the generics like this:

```
withPluginEntities(withTheme<ThemeInterface, Props, typeof undefined>(QueryInput), { completers: 'views.completers' })
```

b) updating the typedefs to v5 fixed this issue and made it work with `withPluginEntities(withTheme(QueryInput)` out of the box. Can you do the typedef update either in this or a separate PR?

Hope this helps.

Reply:
Thanks again for the feedback, I've tested solution b) locally by using https://github.com/Graylog2/graylog2-server/pull/8866 and I am still receiving the flow type error with the `styled-components` v5 interface definition:
```
Error ┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈┈ src/views/components/searchbar/QueryInput.jsx:108:16

Cannot call withPluginEntities because property completers is missing in object type [1] in the first argument.
[prop-missing]

     src/views/components/searchbar/QueryInput.jsx
     105│   placeholder: '',
     106│ };
     107│
     108│ export default withPluginEntities(withTheme(QueryInput), { completers: 'views.completers' });
     109│

     flow-typed/npm/styled-components_v5.x.x.js
 [1] 223│   declare export function withTheme<Theme, Config: { ... }, Instance>(

```
Do you have an idea, why this behaves differently? 

I also Implemented solution a) and realised we need to define explicit types when using `withPluginEntities`. Under these circumstances I prefer the following solution:
```
const mapping = { completers: 'views.completers' };

export default withPluginEntities<$Rest<Props, {| theme: ThemeInterface |}>, typeof mapping>(withTheme(QueryInput), mapping);